lib/WebService/Async/Onfido/RateLimiter.pm

```diff
@@ -0,0 +1,158 @@
+package WebService::Async::Onfido::RateLimiter;
+
+use strict;
+use warnings;
+
+our $VERSION = '0.001';
+
+=head1 NAME
+
+WebService::Async::Onfido::RateLimiter - Module abstract
+
+=head1 SYNOPSIS
+
+    use WebService::Async::Onfido::RateLimiter;
+    my $instance = WebService::Async::Onfido::RateLimiter->new;
+
+=head1 DESCRIPTION
+
+=cut
+
+=head1 METHODS
+
+=cut
+
+use Future;
+use mro;
+
+use parent qw(IO::Async::Notifier);
+
+=head1 NAME
+
+WebService::Async::CustomerIO::RateLimitter - This class provide possibility to limit amount
+of request in time interval
+
+=head1 SYNOPSIS
+
+   use IO::Async::Loop;
+   use WebService::Async::Onfido::RateLimiter;
+   my $loop = IO::Async::Loop->new;
+   $loop->add(
+       my $limiter = WebService::Async::Onfido::RateLimiter->new(
+           interval => 5,
+           limit    => 3
+       ));
+   my $f = $limiter->acquire->then(sub{print "doing "});
+   $loop->run;
+
+=head1 DESCRIPTION
+
+=cut
+
+sub _init {
+    my ($self, $args) = @_;
+    for my $k (qw(limit interval)) {
+        die "Missing required argument: $k" unless exists $args->{$k};
+        die "Invalid value for $k: $args->{$k}" unless int($args->{$k}) > 0;
+        $self->{$k} = delete $args->{$k};
+    }
+
+    $self->{queue}   = [];
+    $self->{counter} = 0;    ## TODO remove ?
+
+    return $self->next::method($args);
+}
+
+=head2 interval
+
+=cut
+
+sub interval { return shift->{interval} }
+
+=head2 limit
+
+=cut
+
+sub limit { return shift->{limit} }
+
+=head2 is_limited
+
+=cut
+
+sub is_limited {
+    my $self = shift;
+    # the number of slots is less.
+    return scalar($self->{queue}->@*) >= $self->limit
+        # the item of [-$limit] is not ready
+        && (
+        !($self->{queue}[-$self->limit]->is_ready)
+        #  or is ready but passed no more than interval seconds
+        || (time() - $self->{queue}[-$self->limit]->get < $self->interval)) ? 1 : 0;
```

Comment:
Could you split this  condition to 3 separate `if statements`  to simplify reading